script.js

```diff
@@ -1,4 +1,210 @@
+    console.log("game mode is changed to player2select");
+    myOutputValue = `<b>WELCOME PLAYER 2 </b>🎲🎲🎲<br><br>You rolled ${player2FirstRollOutput} for Dice 1 and ${player2SecondRollOutput} for Dice 2.<br><br>${selectDiceOrderText}`;
+    return myOutputValue;
+  }
+  // game mode: player 2 to select
+
+  userInput = Number(input);
+  var textToProceedToCheckWinner = "Click submit to find out who won!🧐🤔";
+
+  if (currentGameMode == player2Select) {
+    //input validation for (1) not a number (2) not 1 and not 2
+
+    //for (1)
+    if (Number.isNaN(Number(input)) == true) {
+      myOutputValue = "sorry please enter a number.";
+      return myOutputValue;
+    }
+
+    if (userInput != 1 && userInput != 2) {
+      myOutputValue = "Please insert only either '1' or '2'.";
+      return myOutputValue;
+    }
+
+    if (userInput == 1) {
+      //if input is 1, let the first number go first. Else if input is 2, let second number go first
+      player2Number = String(player2Array[0]) + String(player2Array[1]);
+      myOutputValue = `Player 2, you chose Dice ${userInput} first.<br><br>You rolled ${player2Array}.<br><br>Your number is ${player2Number}.</b><br><br>${textToProceedToCheckWinner}`;
+      // set to the next game mode
+      currentGameMode = matchPlayer1Player2;
+      console.log("game mode is changed to matchplayer1player2");
+      return myOutputValue;
+    } else if (userInput == Number("2")) {
+      player2Number = String(player2Array[1]) + String(player2Array[0]);
+      myOutputValue = `Player 2, you chose Dice ${userInput} first.<br>You rolled ${player2Array}.<br><b>Your number is ${player2Number}.</b><br><br>${textToProceedToCheckWinner}`;
+      // set to the next game mode
+      currentGameMode = matchPlayer1Player2;
+      console.log("game mode is changed to matchplayer1player2");
+      return myOutputValue;
+    }
+  }
+
+  var outputText = `Player 1 chose ${player1Number} while Player 2 chose ${player2Number}.`;
+  // game mode: check who is the winner. Final game mode!
+  if (currentGameMode == matchPlayer1Player2) {
+    //if game mode is matchplayer1player 2, check if numbers are same (draw), Else if then proceed
+
+    if (Number(player1Number) == Number(player2Number)) {
+      myOutputValue = `${outputText}<br>You've gotten a draw. Please try again.`;
+
+      return myOutputValue;
+    } else if (Number(player1Number) > Number(player2Number)) {
+      myOutputValue = `${outputText}<br><br>Player 1 got a larger number than Player2. <br><br><b>Player 1 wins!</b> 🎉🔥🥳<br><br> Click submit to play again.`;
+      isTheGameOver = true; //game is over, hence game is over
+
+      return myOutputValue;
+    } else if (Number(player1Number) < Number(player2Number)) {
+      myOutputValue = `${outputText}<br><br>Player 2 got a larger number than player 1. 
+    <br><br><b>Player 2 wins!</b> 🎉🔥🥳<br><br> Click submit to play again.`;
+      isTheGameOver = true;
+      return myOutputValue;
+    }
+  }
```

Comment:
This is ok, but since you're returning early, you don't need the else ifs and the last condition. 

```
    if (Number(player1Number) == Number(player2Number)) {
      myOutputValue = `${outputText}<br>You've gotten a draw. Please try again.`;

      return myOutputValue;
    }

   if (Number(player1Number) > Number(player2Number)) {
      myOutputValue = `${outputText}<br><br>Player 1 got a larger number than Player2. <br><br><b>Player 1 wins!</b> 🎉🔥🥳<br><br> Click submit to play again.`;
      isTheGameOver = true; //game is over, hence game is over

      return myOutputValue;
    } 

    myOutputValue = `${outputText}<br><br>Player 2 got a larger number than player 1. 
  <br><br><b>Player 2 wins!</b> 🎉🔥🥳<br><br> Click submit to play again.`;
    isTheGameOver = true;
    return myOutputValue;